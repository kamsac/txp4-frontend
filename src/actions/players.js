import dotenvConfiguration from '../dotenv-configuration';
import PlayerResourceMock from '../resources/players/mock';
import PlayerResource from '../resources/players';

export const PLAYER_SUCCESS = 'PLAYER_SUCCESS';
export const PLAYER_REQUEST = 'PLAYER_REQUEST';
export const PLAYER_FAILURE = 'PLAYER_FAILURE';

export function requestPlayer() {
  return {
    type: PLAYER_REQUEST,
    payload: {
      isFetching: true,
    },
  };
}

export function receivePlayer(player) {
  return {
    type: PLAYER_SUCCESS,
    payload: {
      isFetching: false,
      player,
    },
  };
}

export function playerError() {
  return {
    type: PLAYER_FAILURE,
    payload: {
      isFetching: false,
    },
  };
}

export function loadPlayer(playerLogin) {
  return (dispatch) => {
    dispatch(requestPlayer());

    const resource = dotenvConfiguration.API_URL ? PlayerResource : PlayerResourceMock;

    resource.getPlayer(playerLogin)
      .then((response) => {
        dispatch(receivePlayer(response.data.player));
      })
      .catch(() => {
        dispatch(playerError());
      });
  };
}
