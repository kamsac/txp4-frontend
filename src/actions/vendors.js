import VendorsResource from '../resources/vendors/mock';

export const REQUEST_VENDORS = 'vendors:requestVendors';
export function requestVendors() {
  return dispatch => {
    return dispatch(fetchVendors());
  }
}

function fetchVendors() {
  return dispatch => {
    VendorsResource.getVendors((vendors) => {
      dispatch(receiveVendors(vendors));
    });
  }
}

export const RECEIVE_VENDORS = 'vendors:fetchVendorsSuccess';
export function receiveVendors(vendors) {
  return {
    type: RECEIVE_VENDORS,
    vendors: vendors,
  }
}
