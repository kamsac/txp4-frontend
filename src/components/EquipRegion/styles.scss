@import '../../variables';

.EquipRegion {
  display: grid;
  grid-gap: $grid-gap-small/2;
  grid-template-areas:
    "InventoryItem"
    "EquippedInventoryItemDetails";
  grid-template-columns: 1fr;
  grid-template-rows: 100px 1fr;
}
