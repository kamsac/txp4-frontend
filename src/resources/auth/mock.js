export const ACCESS_TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjoxNTE2MjM5MDIyLCJsb2dpbiI6Imx5bWFrIiwibmljayI6IiRzJG8kYTYwbCRjODJ5JGZhM20kYzgyYSRhNjBrIn0.aaKzXEZP2tovhBAqPJYBdmbLzKjfdBsuX3FaB58JZvE';

const AuthResourceMock = {
  login() {
    return new Promise((resolve) => {
      const response = {
        data: {
          token: ACCESS_TOKEN,
        },
      };
      setTimeout(() => resolve(response), 100);
    });
  },
};

export default AuthResourceMock;
