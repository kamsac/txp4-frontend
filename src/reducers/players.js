import inventoryReducer from './inventory';
import { PLAYER_FAILURE, PLAYER_REQUEST, PLAYER_SUCCESS } from '../actions/players';
import {
  INVENTORY_REQUEST,
  INVENTORY_FAILURE,
  INVENTORY_SUCCESS,
  EQUIP_REQUEST,
  EQUIP_SUCCESS,
  EQUIP_FAILURE,
} from '../actions/inventory';

const initialState = {
  isFetching: false,
  items: {},
};

export default function playersReducer(state = initialState, action) {
  switch (action.type) {
    case PLAYER_REQUEST: {
      return Object.assign({}, state, {
        isFetching: action.payload.isFetching,
      });
    }
    case PLAYER_SUCCESS: {
      const { isFetching, player } = action.payload;
      return Object.assign({}, state, {
        isFetching,
        items: Object.assign({}, state.items, {
          [player.login]: {
            login: player.login,
            nick: player.nick,
            inventory: {
              isFetching: false,
              value: player.inventory,
            },
          },
        }),
      });
    }
    case PLAYER_FAILURE: {
      const { isFetching } = action.payload;
      return Object.assign({}, state, {
        isFetching,
      });
    }

    case INVENTORY_REQUEST:
    case INVENTORY_SUCCESS:
    case INVENTORY_FAILURE:
    case EQUIP_REQUEST:
    case EQUIP_SUCCESS:
    case EQUIP_FAILURE: {
      const { playerLogin } = action.payload;

      const nextState = Object.assign({}, state);

      if (!nextState.items[playerLogin]) {
        nextState.items[playerLogin] = {};
      }

      const inventory = inventoryReducer(nextState.items[playerLogin].inventory, action);
      nextState.items[playerLogin].inventory = inventory;
      return nextState;
    }
    default:
      return state;
  }
}

export function getPlayer(state, playerLogin) {
  if (
    state.players &&
    state.players.items
  ) {
    return state.players.items[playerLogin];
  }

  return null;
}

export function getPlayerNick(state, playerLogin) {
  if (
    state.players &&
    state.players.items &&
    state.players.items[playerLogin]
  ) {
    return state.players.items[playerLogin].nick;
  }

  return null;
}
