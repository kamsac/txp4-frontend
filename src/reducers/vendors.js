import {RECEIVE_VENDORS} from '../actions/vendors';

const initialValue = [];

export function vendorsReducer(state = initialValue, action) {
  switch(action.type) {
    case RECEIVE_VENDORS:
      return [...action.vendors];
    default:
      return state;
  }
}
