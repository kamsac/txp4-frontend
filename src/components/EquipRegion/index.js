import React from 'react';
import './styles.scss';
import InventoryItem from '../InventoryItem';
import EquippedInventoryItemDetails from '../EquippedItemDetails';

export default class EquipRegion extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="EquipRegion">
        <InventoryItem item={this.props.item} />
        <EquippedInventoryItemDetails item={this.props.item} />
      </div>
    )
  }
}
