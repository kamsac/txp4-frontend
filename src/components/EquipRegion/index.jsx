import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';
import InventoryItem from '../InventoryItem/index';
import EquippedItemDetails from '../EquippedItemDetails/index';
import { ItemPropTypesShape, VendorPropTypesShape } from '../../prop-types';

export default class EquipRegion extends React.Component {
  render() {
    return (
      <div className="EquipRegion">
        <InventoryItem item={this.props.item} vendors={this.props.vendors} />
        <EquippedItemDetails item={this.props.item} vendors={this.props.vendors} />
      </div>
    );
  }
}

EquipRegion.propTypes = {
  item: ItemPropTypesShape.isRequired,
  vendors: PropTypes.arrayOf(VendorPropTypesShape).isRequired,
};
