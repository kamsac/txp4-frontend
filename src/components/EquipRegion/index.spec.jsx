import React from 'react';
import { shallow } from 'enzyme';
import EquipRegion from './index';
import { ITEMS } from '../../resources/players/mock';
import { VENDORS } from '../../resources/vendors/mock';

test('<EquipRegion>', () => {
  const component = shallow(<EquipRegion item={ITEMS[0]} vendors={VENDORS} />);
  expect(component).toMatchSnapshot();
});
