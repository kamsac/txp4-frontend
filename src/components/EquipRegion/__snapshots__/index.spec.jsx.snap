// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EquipRegion> 1`] = `
<div
  className="EquipRegion"
>
  <InventoryItem
    item={
      Object {
        "equipRegion": "tire",
        "id": "a1a1a1",
        "isEquipped": true,
        "modifiers": Array [
          Object {
            "name": "maxSpeed",
            "value": 1,
          },
          Object {
            "name": "acceleration",
            "value": 1.2,
          },
          Object {
            "name": "steering",
            "value": 1.1,
          },
          Object {
            "name": "breaking",
            "value": 1.2,
          },
          Object {
            "name": "grip",
            "value": 1.5,
          },
          Object {
            "name": "gravity",
            "value": 1,
          },
        ],
        "name": "Tire",
        "tier": 3,
        "vendorId": "aaa",
      }
    }
    onSelectItem={[Function]}
    selectedItem={null}
    vendors={
      Array [
        Object {
          "color": "#8f0",
          "id": "aaa",
          "name": "Super",
        },
        Object {
          "color": "#f80",
          "id": "bbb",
          "name": "Safe",
        },
        Object {
          "color": "#800",
          "id": "ccc",
          "name": "Heavy",
        },
        Object {
          "color": "#08f",
          "id": "ddd",
          "name": "Light",
        },
      ]
    }
  />
  <EquippedItemDetails
    item={
      Object {
        "equipRegion": "tire",
        "id": "a1a1a1",
        "isEquipped": true,
        "modifiers": Array [
          Object {
            "name": "maxSpeed",
            "value": 1,
          },
          Object {
            "name": "acceleration",
            "value": 1.2,
          },
          Object {
            "name": "steering",
            "value": 1.1,
          },
          Object {
            "name": "breaking",
            "value": 1.2,
          },
          Object {
            "name": "grip",
            "value": 1.5,
          },
          Object {
            "name": "gravity",
            "value": 1,
          },
        ],
        "name": "Tire",
        "tier": 3,
        "vendorId": "aaa",
      }
    }
    vendors={
      Array [
        Object {
          "color": "#8f0",
          "id": "aaa",
          "name": "Super",
        },
        Object {
          "color": "#f80",
          "id": "bbb",
          "name": "Safe",
        },
        Object {
          "color": "#800",
          "id": "ccc",
          "name": "Heavy",
        },
        Object {
          "color": "#08f",
          "id": "ddd",
          "name": "Light",
        },
      ]
    }
  />
</div>
`;
